mantidproject/mantid
# Fit Script Generator - Generate Sequential Fit Python Script

**Description of work.**
This PR adds an algorithm `GeneratePythonFitScript` that can be used to generate a python script used for sequential fitting.

This algorithm has also been connected up to the `Fit Script Generator` interface so that you can generate a sequential fitting python script and save it to a file, or copy the generated script to the clipboard.

In a future PR this algorithm will be expanded to generate a python script used for sequential fitting.

**To test:**
1. Comment out this line
https://github.com/mantidproject/mantid/blob/44daa7417226a56f8d1896ea6b2f2b1bc576a5d1/scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_view.py#L52
2. Go to Muon Analysis interface
3. Make sure the `MUSR` instrument is selected. Type in run 62261 and wait for it to load
4. Go to Fitting tab and click `Fit Generator`. There should be 4 data domains in the table
6. Right click the function browser and add a `GausOsc`
7. Change the frequency parameter to 1
8. At this point all the data you need is set up, you can now start testing what this PR adds:

- You can try generating a script to a file using `Generate Fit Script to File`
- You can try generating a script to clipboard using `Generate Fit Script to Clipboard`

9. After doing one of these either load the python script into mantid or paste it into a script from the clipboard.
10. Run the Script and you should get 4 fits plotted side-by-side

11. Read the python script and make sure it is clear to read/understand for a typical user

Fixes #29911

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
